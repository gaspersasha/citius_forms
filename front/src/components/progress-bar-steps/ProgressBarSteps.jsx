import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import s from './progress-bar-steps.module.sass';

export const ProgressBarSteps = ({ steps, currentStep, progress }) => {
  const currentStepIndex = steps.findIndex((step) => step.id === currentStep);

  return (
    <div className={s.container}>
      <div className={s.bar}>
        {steps.map((step, index) => (
          <Fragment key={step.id}>
            {currentStepIndex > index && (
              <div
                className={`${s.step} ${s.passed}`}
                data-id="progress-bar-step"
                key={step.id}
              >
                <div className={s.icon} />
                <div className={s.title}>{step.text}</div>
              </div>
            )}
            {currentStepIndex === index && (
              <div
                className={`${s.step} ${s.current}`}
                data-id="progress-bar-step"
                key={step.id}
              >
                <div className={s.icon}>
                  <div className={s.diagram}>
                    <svg
                      width="51px"
                      height="52px"
                      viewBox="0 0 42 42"
                      className="donut"
                    >
                      <circle
                        className="donut-segment"
                        cx="21"
                        cy="21"
                        r="15.91549430918954"
                        fill="transparent"
                        stroke="#0076FD"
                        strokeWidth="3"
                        strokeDasharray={`${progress} ${100 - progress}`}
                        strokeDashoffset="25"
                        strokeLinecap="round"
                      />
                    </svg>
                  </div>
                  <span className={s.progress}>{`${progress}%`}</span>
                </div>
                <div className={s.title}>{step.text}</div>
              </div>
            )}
            {currentStepIndex < index && (
              <div
                className={`${s.step} ${s.future}`}
                data-id="progress-bar-step"
                key={step.id}
              >
                <div className={s.icon}>
                  <span className={s.stepNumber}>{index + 1}</span>
                </div>
                <div className={s.title}>{step.text}</div>
              </div>
            )}
          </Fragment>
        ))}
      </div>
    </div>
  );
};

ProgressBarSteps.propTypes = {
  steps: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
    })
  ).isRequired,
  currentStep: PropTypes.string.isRequired,
  progress: PropTypes.number.isRequired,
};

export default ProgressBarSteps;
