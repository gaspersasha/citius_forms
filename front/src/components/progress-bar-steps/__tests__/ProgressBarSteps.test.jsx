import React from 'react';
import { ProgressBarSteps } from '../ProgressBarSteps';

describe('<ProgressBarSteps />', () => {
  let wrapper;

  const getWrapper = (props) => shallow(<ProgressBarSteps {...props} />);

  const config = {
    steps: ['Step1', 'Step2', 'Step3', 'Step4', 'Step5'],
    currentStep: 'Step3',
    progress: 88,
  };

  it('renders without errors', () => {
    wrapper = getWrapper(config);
    expect(wrapper).toMatchSnapshot();
  });

  it('render steps', () => {
    wrapper = getWrapper(config);
    expect(wrapper.find('[data-id="progress-bar-step"]').length).toEqual(
      config.steps.length
    );
  });
});
