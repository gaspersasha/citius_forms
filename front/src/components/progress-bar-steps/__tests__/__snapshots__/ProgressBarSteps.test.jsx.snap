// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBarSteps /> renders without errors 1`] = `
<div
  className="container"
>
  <div
    className="bar"
  >
    <div
      className="step future"
      data-id="progress-bar-step"
    >
      <div
        className="icon"
      >
        <span
          className="stepNumber"
        >
          1
        </span>
      </div>
      <div
        className="title"
      />
    </div>
    <div
      className="step future"
      data-id="progress-bar-step"
    >
      <div
        className="icon"
      >
        <span
          className="stepNumber"
        >
          2
        </span>
      </div>
      <div
        className="title"
      />
    </div>
    <div
      className="step future"
      data-id="progress-bar-step"
    >
      <div
        className="icon"
      >
        <span
          className="stepNumber"
        >
          3
        </span>
      </div>
      <div
        className="title"
      />
    </div>
    <div
      className="step future"
      data-id="progress-bar-step"
    >
      <div
        className="icon"
      >
        <span
          className="stepNumber"
        >
          4
        </span>
      </div>
      <div
        className="title"
      />
    </div>
    <div
      className="step future"
      data-id="progress-bar-step"
    >
      <div
        className="icon"
      >
        <span
          className="stepNumber"
        >
          5
        </span>
      </div>
      <div
        className="title"
      />
    </div>
  </div>
</div>
`;
